core/src/main/java/com/oath/oak/OakWBuffer.java

```diff
@@ -0,0 +1,252 @@
+/**
+ * Copyright 2018 Oath Inc.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.oath.oak;
+
+import java.nio.*;
+import java.util.function.Function;
+
+/**
+ * A similar to ByteBuffer interface that allows internal Oak data access for read and write
+ */
+public interface OakWBuffer {
+
+    /**
+     * Returns this buffer's capacity.
+     *
+     * @return The capacity of this buffer
+     */
+    int capacity() throws NullPointerException;
+
+    /**
+     * Going to be deprecated!!!!!!!!!
+     *
+     * @return the actual ByteBuffer
+     */
+    @Deprecated
+    ByteBuffer getByteBuffer();
+
+    // -- Singleton get/put methods --
+
+    /**
+     * Absolute <i>get</i> method.  Reads the byte at the given
+     * index.
+     *
+     * @param index The index from which the byte will be read
+     * @return The byte at the given index
+     * @throws IndexOutOfBoundsException If <tt>index</tt> is negative
+     *                                   or not smaller than the buffer's limit
+     */
+    byte get(int index) throws NullPointerException;
```

Comment:
I see both in (R and in W buffers) only absolute get and put methods (getting an index for position). What about relative get and put methods (getting nothing and using internal position)? This should be allowed only in W buffer as it updates the internals, but this is a valid part of ByteBuffer interface. Do I miss something?

Reply:
@yonigottesman changed the underlying implementation to avoid extra ByteBuffer slicing. The application now cannot assume position is 0 and it cannot modify the position. 

I also updated the README to state OakBuffers expose an API *similar* to the standard Java API.